dev-tools/mage/deps.go

```diff
@@ -0,0 +1,25 @@
+package mage
+
+import (
+	"fmt"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+// Deps contains targets related to checking dependencies
+type Deps mg.Namespace
+
+// CheckModuleTidy checks if `go mod tidy` was run before the last commit.
+func (Deps) CheckModuleTidy() error {
```

Comment:
What if instead of running `go mod` with `sh.Run` we used the package https://github.com/elastic/beats/tree/main/dev-tools/mage/gotool? I started porting a few calls because I need it for licence checking and for generating the NOTICE file.

Reply:
I'm fine with that but I'd rather do it as a separate PR since this does not change the logic, it's just moving to a separate package.